app/javascript/courses/review/components/CoursesReview__GradeCard.re

```diff
@@ -52,35 +69,40 @@ module UndoGradingMutation = [%graphql
   |}
 ];
 
-let undoGrading = (submissionId, setState) => {
-  setState(state => {...state, saving: true});
+let undoGrading = (submissionId, send) => {
+  send(BeginSaving);
 
   UndoGradingMutation.make(~submissionId, ())
   |> GraphqlQuery.sendQuery
   |> Js.Promise.then_(response => {
        response##undoGrading##success
-         ? DomUtils.reload() |> ignore
-         : setState(state => {...state, saving: false});
+         ? DomUtils.reload() |> ignore : send(FinishSaving);
        Js.Promise.resolve();
      })
   |> ignore;
 };
 
+let trimToOption = s =>
+  switch (s |> String.trim) {
+  | "" => None
+  | s => Some(s)
+  };
+
```

Comment:
Should we consider moving this to a utils module? It seems like a common pattern that would come in use in future.

Reply:
Nah, let's wait and see if we repeat this elsewhere.